ci/docker/conda-python.dockerfile

```diff
@@ -22,10 +22,12 @@ FROM ${repo}:${arch}-conda-cpp
 # install python specific packages
 ARG python=3.8
 COPY ci/conda_env_python.txt \
+     ci/conda_env_python_unix.txt \
      ci/conda_env_sphinx.txt \
      /arrow/ci/
 RUN mamba install -q -y \
         --file arrow/ci/conda_env_python.txt \
+        --file arrow/ci/conda_env_python_unix.txt \
```

Comment:
Maybe just add gdb directly here as it is not used anywhere else? Should we add more packages/use them in multiple location we can always refactor into a new file.

Reply:
I think that this approach is better. See also: https://github.com/apache/arrow/pull/35024#issuecomment-1504133502

But it's not a strong opinion because I'm not a conda user. I follow opinions from developers who use conda.